src/components/dialog/dialog.atoms.tsx

```diff
@@ -0,0 +1,36 @@
+import { forwardRef } from 'react'
+
+import { DialogBodyProps, DialogFooterProps, DialogContainerProps, DialogTitleProps } from './types'
+import { ElDialogBody, ElDialogContainer, ElDialogFooter, ElDialogTitle } from './style'
+
+const Container = forwardRef<HTMLDialogElement, DialogContainerProps>(({ children, ...rest }, ref) => {
+  return (
+    <ElDialogContainer {...rest} ref={ref}>
+      {children}
+    </ElDialogContainer>
+  )
+})
+
+const Title = ({ title, ...rest }: DialogTitleProps) => (
```

Comment:
> I view the "title" as being the text that titles the dialog, not the actual atom. I think the atom here is a "header" because it may include more than just the title in future (e.g. an "X" button to dismiss the dialog)

That's acceptable, thanks

> Can you please help me understand why you're passing title to the ElDialogTitle components title attribute

If I'm not mistaken, the dialog need to have the `title` attribute that will placed in or within the Dialog atoms. Looks like in the past I saw a warning from a11y storybook plugin, then passing a title attribute is clearing the issue 🤔

> This is because <Dialog.Title>My Title</Dialog.Title> feels more natural/HTML-y than <Dialog.Title title="My Title" />

Agree on your thought, keeping to use the same pattern as we can.

**thought:** Talking about your first suggestion looks like we might only pass the `string` type instead `ReactNode`, so the extendable things in the future might not be able to achieve with that atom. 

**suggestion:** Following my thought above, make a separate between `Dialog.Header` and `Dialog.Title` will make them smile? the `Dialog.Title` atoms should used under `Dialog.Header`, and the `Dialog.Title` is only acceptable if we're pass a string data

```tsx
<Dialog>
  <Dialog.Header>
    <Dialog.Title>foo</Dialog.Title>
    <AnotherHeaderComponent />
  </Dialog.Header>
</Dialog>
```

Reply:
> Regarding the need for a title attribute, I don't see anything in [MDN's docs on the element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) that mentions that requirement. Is that coming from an Elements/Storybook a11y plugin or somewhere else?

the storybook a11y plugin mention the warning at that time; but with upcoming commit the warn is vanished